### ISSUE TITLE
CGAL intersection bug

### ISSUE THREAD

**Issue by jh66637** [dealii/dealii#14359]
@fdrmrc  While introducing a tolerance for intersections (https://github.com/dealii/dealii/pull/14353), I might have found a bug. Maby this is a CGAL bug (I am using version 5.5).

I am trying to find the intersection between a line and a rectangular cell. While there is no chance in finding the intersection without rounding the vertex locations, I already used the tolerance from https://github.com/dealii/dealii/pull/14353 to ensure an intersection and end up with a rectangle that exactly contains the line I am searching for. Thus, the amount of found intersections should be 1.

On my machine, no intersection is found. 

I searched quite some time now where this could come from, but I am currently stuck.

```  
  Triangulation<2,2> tria0;
  Triangulation<1,2> tria1;
  GridGenerator::hyper_cube(tria0);
  GridGenerator::hyper_cube(tria1);

  const auto cell0 = tria0.begin_active();
  const auto cell1 = tria1.begin_active();
  cell0->vertex(1)=Point<2>(1.0000000000000000555111512e-01, 3.3799999999999996713739847e-02);
  cell0->vertex(2)=Point<2>(1.0000000000000000555111512e-01, 6.7100000000000006861178292e-02);

  // Vertices are exactly the same on one side of the rectange and the line 
  cell0->vertex(0)=Point<2>(6.7100000000000006861178292e-02, 3.3799999999999996713739847e-02);
  cell1->vertex(0)=Point<2>(6.7100000000000006861178292e-02, 3.3799999999999996713739847e-02);

  cell0->vertex(3)=Point<2>(6.7100000000000006861178292e-02, 6.7100000000000006861178292e-02);
  cell1->vertex(1)=Point<2>(6.7100000000000006861178292e-02, 6.7100000000000006861178292e-02);

  const auto vec_of_arrays =
    CGALWrappers::compute_intersection_of_cells<2, 1, 2>(cell0, cell1, MappingQ1<2,2>(), MappingQ1<1,2>(), 1e-9);

  assert(vec_of_arrays.size()==1);
```

**Comment by fdrmrc** [comment]
The problem is that the skeleton of the 2d cell is as in picture. `vertex(2)` and `vertex(3)` for `tria0` are swapped. If you swap them, the intersection is correctly found. 

![johannes_issue](https://user-images.githubusercontent.com/50499769/196496897-3eb1ad01-5e87-489d-b162-5dc87d4cecab.png)

**Comment by fdrmrc** [comment]
You're welcome! Sure, keep me posted.

**Comment by jh66637** [comment]
The problem I had was that I used the vertex version of ```compute_intersection_of_cells()```. This triggered the function definition in the header instead of an explicit instantiation. I couldn't see the assert of the header function definition because I aborted the program before. When trying to reproduce, I used the iterator version of the same function and forgot that the vertices I obtained before were already ordered for CGAL. 

https://github.com/dealii/dealii/pull/14362 fixes this. I will close the issue once it's merged.
